src/modules/Builder/features/ActivityItems/ItemConfiguration/GroupedSelectSearchController/GroupedSelectSearchController.tsx

```diff
@@ -120,6 +121,12 @@ export const GroupedSelectSearchController = <T extends FieldValues>({
         control={control}
         render={({ field: { onChange, value }, fieldState: { error } }) => {
           const handleOnSelectChange = (...props: unknown[]) => {
+            const newValue = (props[0] as { target: { value: ItemResponseType } }).target.value;
```

Comment:
Nit: I find the syntax here a little awkward to read. What do you think of using stronger typing in the event handler? It would also require updating `onChange()` calls below, but simplifies the `processItemType()` call…
```suggestion
          const handleOnSelectChange = (event: ChangeEvent<HTMLInputElement>, child: ReactNode) => {
            const newValue = event.target.value as ItemResponseType;
```

Reply:
Updating 🤓

Reply:
Thanks @andrevitalb! Just to ensure there's no regression risk, I think it'd be safest to continue passing the second argument being passed to `handleOnSelectChange` – the `ReactNode` variable – to the `onChange` handler in both calls to it, as it was originally receiving both arguments.

Reply:
Oh, totally my bad, meant to include it 😅. Updated